src/bentoml/_internal/cloud/deployment.py

```diff
@@ -106,7 +106,7 @@ def verify(
                     ("bento", self.bento),
                     ("cluster", self.cluster),
                     ("access_authorization", self.access_authorization),
-                    ("envs", self.envs if len(self.envs) > 0 else None),
+                    ("envs", self.envs if self.envs and len(self.envs) > 0 else None),
```

Comment:
```suggestion
                    ("envs", self.envs if self.envs else None),
```

Reply:
We need to make sure `envs` is not an empty array.

Reply:
`self.envs` is also falsy for empty array